api/health.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+import (
+	"errors"
+	"net/http"
+	"sync"
+
+	"github.com/hashicorp/consul-terraform-sync/api/oapigen"
+	"github.com/hashicorp/consul-terraform-sync/health"
+	"github.com/hashicorp/consul-terraform-sync/logging"
+)
+
+const (
+	getHealthSubsystemName = "gethealth"
+)
+
+// HealthHandler handles the health endpoints
+type HealthHandler struct {
+	mu sync.RWMutex
+	hm health.Manager
+}
+
+// NewHealthHandler creates a new health handler using the provided health manager
+// to determine health
+func NewHealthHandler(hm health.Manager) *HealthHandler {
+	return &HealthHandler{
+		hm: hm,
+	}
+}
+
+// GetHealth returns the health status
+func (hh *HealthHandler) GetHealth(w http.ResponseWriter, r *http.Request) {
+	hh.mu.RLock()
+	defer hh.mu.RUnlock()
+
+	logger := logging.FromContext(r.Context()).Named(getHealthSubsystemName)
+	logger.Trace("get health")
+
+	err := hh.hm.Check()
```

Comment:
Now I'm leaning towards returning an "isHealthy" boolean check over the error type :S

Reply:
I think either would be fine, especially since we're realistically never going to return an error for now! I think the main crux of the decision would be whether we want to treat unhealthy different from not being able to determine the health. 

It may even be that you can return both a boolean and an error and be able to distinguish between the two that way. `false, nil` being we successfully checked the health and it's unhealthy, and `false, "some error"` could be for errors checking the health.

Reply:
I'll just leave it as an error type for now, we can revisit when we add more robust health checking